src/html/classic/ng/src/gmp/commands/trashcan.js

```diff
@@ -0,0 +1,61 @@
+/* Greenbone Security Assistant
+ *
+ * Authors:
+ * Steffen Waterkamp <steffen.waterkamp@greenbone.net>
+ *
+ * Copyright:
+ * Copyright (C) 2017 Greenbone Networks GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import {HttpCommand, register_command} from '../command.js';
+
+class Trashcan extends HttpCommand {
+
+  restore({id}) {
+    const data = {
+      cmd: 'restore',
+      target_id: id,
+    };
+    return this.httpPost(data);
+  }
+
+  delete({id, entity_type}) {
+    const command = 'delete_trash_' + entity_type;
+    const type_id = entity_type + '_id';
+    const data = {
+      cmd: [command],
+      [type_id]: id,
+    };
+    return this.httpPost(data);
+  }
+
+  emptyTrashcan() {
```

Comment:
Maybe rename it to only empty() ? The api will be gmp.trashcan.empty() which is nicer then gmp.tashcan.emptyTashcan()

Maybe also gmp.tashcan.flush ?

Reply:
Changed it to "empty()" in order to stay closer to the original naming of "empty_trashcan" http-command.